provider/pkg/await/condition/source.go

```diff
@@ -95,16 +95,13 @@ func (des *DynamicSource) Start(_ context.Context, gvk schema.GroupVersionKind)
 	}
 	i := informer.Informer()
 
-	// client-go logs a warning if we've already started the informer for this
-	// GVK, but the method to check whether that's the case isn't part of the
-	// public interface.
-	if check, ok := i.(interface{ HasStarted() bool }); ok {
-		if check.HasStarted() {
-			return events, nil
-		}
-	}
-	go informer.Informer().Run(des.stopper)
-	factory.WaitForCacheSync(des.stopper)
+	// Start the new informer by calling factory.Start (which is idempotent).
+	// This ensures that the informer is started exactly once and is cleaned up later.
+	factory.Start(des.stopper)
```

Comment:
See implementation of [factory.Start](https://github.com/kubernetes/client-go/blob/c106b23895edc59ff05c65770a17e6a6d3caee66/dynamic/dynamicinformer/informer.go#L91-L113)